### PR TITLE
Enable sync job to be called from the management api

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
## Related Issues and Dependencies

Partially addresses: https://github.com/thoth-station/thoth-application/issues/2513 (task 3 / 4)
Related-to: https://github.com/thoth-station/management-api/issues/887

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

New post endpoints for `/graph/sync-job/{component-name}`, that will trigger the sync job for those documents.
Component names:
1. adviser
2. solver
3. revsolver
4. package-extract
5. provenance-checker
6. security-indicators

/cc @harshad16 @fridex

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -1691,6 +1691,156 @@ paths:
         "401":
           description: On invalid secret.
 
+  /graph/sync-job/solver:
```
Comment (fridex) [inline openapi/openapi.yaml:1694]: Maybe could be parametrized with an enum?

```suggestion
  /graph/sync-job/{type}:
```

File: thoth/management_api/api_v1.py [hunk 2]
```diff
@@ -61,6 +67,102 @@ def get_info(secret: str):
...
+def post_sync_job_solver_documents(
+    force: bool = False,
+    graceful: bool = False,
+) -> tuple:
+    """Tringger the Sync Job for Solver documents."""
+    from .openapi_server import GRAPH
+
+    try:
+        sync_solver_documents(
+            force=force, graph=GRAPH, graceful=graceful, is_local=False
```
Comment (fridex) [inline thoth/management_api/api_v1.py:79]: The HTTP server that we run has multiple workers that try to serve requests. Each worker has a timeout for requests to be served. If the worker is not able to deliver any response, it is killed. Considering we will sync thousands of documents, this job needs to be done asynchronously as a job in the cluster.

We have routies in thoth-common that can submit jobs to the cluster using the OpenShift abstraction. In that case, the sync will happen in the cluster asynchronously to the request and will not timeout.

File: thoth/management_api/sync_job_types.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+
+"""Graph Sync Job Document types."""
+
+class document_type(Enum):
```
Comment (harshad16) [inline thoth/management_api/sync_job_types.py:22]: can we move this under the function `schedule_sync_job`?
i don't see use of this anywhere else apart from schedule_sync_job

File: openapi/openapi.yaml [hunk 4]
```diff
@@ -1670,6 +1670,43 @@ paths:
...
+      operationId: schedule_sync_job
+      summary: Trigger the Sync job for solver documents
+      parameters:
+        - name: secret
+          in: query
+          required: true
+          description: A secret to schedule the job.
+          schema:
+            type: string
+        - name: document_type
```
Comment (harshad16) [inline openapi/openapi.yaml:1686]: can we do the enums with some description of the enum options and 
schema with enums ?

```
description: type of documents ...
Advise: ...
solver:..
..
schema:
  type: string
  enum: [advise,solver,etc]
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
Great effort :tada: 

Is this been tested ? 
does the change of the open API version cause any issue with the existing schema ?

File: openapi/openapi.yaml (hunk 4 above)
Comment (Gregory-Pereira) [reply openapi/openapi.yaml:1686]: Not certain about what type of descriptions we would have for the enum options, its simply that we would be syncing those types of documents. Let me know if this does not work.

Comment (harshad16) [reply openapi/openapi.yaml:1686]: Here is the documents: https://swagger.io/docs/specification/data-models/enums/

File: openapi/openapi.yaml [hunk 5]
```diff
@@ -1670,6 +1670,57 @@ paths:
         "500":
           description: On server side errors.
 
+  /graph/sync-job/{docuement_type}:
```
Comment (harshad16) [inline openapi/openapi.yaml:1673]:
```suggestion
  /graph/sync-job:
```
I think this is not needed anymore.

File: thoth/management_api/api_v1.py [hunk 6]
```diff
@@ -327,6 +329,32 @@ def schedule_graph_refresh(secret: str):
...
+):
+    """Schedule the sync job for a document type."""
+
+    class SyncJobDocumentType(Enum):
+        solver = 0
+        adviser = 1
+        revsolver = 2
+        package_extract = 3
+        provenance_checker = 4
+        security_indicators = 5
```
Comment (harshad16) [inline thoth/management_api/api_v1.py:346]: why is this required.
seems like sync_job take the value as string
https://github.com/thoth-station/sync-job/blob/9efe989935f77df47cd2e9ee4efee21c0bfa088a/app.py#L46
https://github.com/thoth-station/storages/blob/7d01189226ebac008f8cb13553dd72b186ff31c0/thoth/storages/sync.py#L557

File: thoth/management_api/api_v1.py [hunk 7]
```diff
@@ -327,6 +329,32 @@ def schedule_graph_refresh(secret: str):
...
+
+    if secret != Configuration.THOTH_MANAGEMENT_API_TOKEN:
+        return {"error": "Wrong secret provided"}, 401
+    workflow_id = _OPENSHIFT.schedule_sync_job(
+        document_type=SyncJobDocumentType(document_type),
```
Comment (harshad16) [inline thoth/management_api/api_v1.py:351]:
```suggestion
        document_type=document_type,
```

File: openapi/openapi.yaml [hunk 8]
```diff
@@ -1,4 +1,4 @@
-openapi: "3.0.0"
+openapi: "3.1.2"
```
Comment (harshad16) [inline openapi/openapi.yaml:1]:
```suggestion
openapi: "3.0.0"
```
enum is available in openapi 3.0, so let's keep that.

**Review by harshad16 — CHANGES_REQUESTED** [review]
thanks for all the changes
some more suggestions.

**Review by harshad16 — CHANGES_REQUESTED** [review]
some more changes required

**Review by harshad16 — APPROVED** [review]
/lgtm
thanks :100:

**Pull request merged**